### PR TITLE
Crate database entity

### PR DESCRIPTION
**Pull request by sjcorbett**
An entity suitable for single-node deployments of Crate. Subsequent work should support clusters as described by the docs: https://crate.io/docs/en/latest/best_practice/multi_node_setup.html.

### CODE REVIEW

**Comment by sjcorbett** [comment]
@adamecjackson Thanks for helping out.

To the reviewer: Adam has submitted an ICLA but I don't think it has been filed yet.

**Comment by sjcorbett** [comment]
ICLA now filed.

File: software/database/src/main/java/brooklyn/entity/database/crate/CrateNode.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    public static final PortAttributeSensorAndConfigKey CRATE_HTTP_PORT = new PortAttributeSensorAndConfigKey(
+            "crate.httpPort", "The port for HTTP traffic", PortRanges.fromString("4200+"));
+
+    AttributeSensor<String> MANAGEMENT_URL = Sensors.newStringSensor(
+            "crate.managementUri", "The address at which the Crate server listens");
+
+    AttributeSensor<String> SERVER_NAME = Sensors.newStringSensor(
+            "crate.server.name", "The name of the server");
+
+    AttributeSensor<Boolean> SERVER_OK = Sensors.newBooleanSensor(
```
Comment (andreaturli) [inline software/database/src/main/java/brooklyn/entity/database/crate/CrateNode.java]: what is the difference between SERVER_OK and SERVER_STATUS? are both needed?

Comment (sjcorbett) [inline software/database/src/main/java/brooklyn/entity/database/crate/CrateNode.java]: No idea. They're both reported by Crate so we included both.

**Comment by andreaturli** [comment]
lgtm +1

good stuff, @adamecjackson and @sjcorbett

**Pull request merged**